atdgen/src/json.ml

```diff
@@ -62,9 +86,38 @@ let get_json_float an : json_float =
 
 let json_list_of_string s : json_list option =
   match s with
-      "array" -> Some Array
-    | "object" -> Some Object
-    | _ -> None
+  | "array" -> Some Array
+  | "object" -> Some Object
+  | _ -> (* error *) None
+
+(*
+   <json adapter.ocaml="Foo.Bar">
+   --> { ocaml_adapter = Some "Foo.Bar";
+         java_adapter = None; }
+*)
+let get_json_adapter an =
+  let ocaml_adapter =
+    Atd.Annot.get_field (fun s -> Some (Some s))
+      None ["json"] "adapter.ocaml" an
+  in
+  let java_adapter =
+    Atd.Annot.get_field (fun s -> Some (Some s))
```

Comment:
Can you remind me why we need the double option type here (and other places)?

Reply:
I know it's difficult to make sense of it. I'll try to clarify this with commented examples.

Reply:
I created `Atd.Annot.get_opt_field` and added labeled arguments for clarity.